setuptools/build_meta.py

```diff
@@ -145,18 +151,18 @@ def suppress_known_deprecation():
         yield
 
 
-_ConfigSettings: TypeAlias = Union[Dict[str, Union[str, List[str], None]], None]
+_ConfigSettings: TypeAlias = Union[Mapping[str, Union[str, List[str], None]], None]
 """
 Currently the user can run::
 
     pip install -e . --config-settings key=value
     python -m build -C--key=value -C key=value
 
 - pip will pass both key and value as strings and overwriting repeated keys
-  (pypa/pip#11059).
+(pypa/pip#11059).
 - build will accumulate values associated with repeated keys in a list.
-  It will also accept keys with no associated value.
-  This means that an option passed by build can be ``str | list[str] | None``.
+It will also accept keys with no associated value.
+This means that an option passed by build can be ``str | list[str] | None``.
```

Comment:
```suggestion
  It will also accept keys with no associated value.
  This means that an option passed by build can be ``str | list[str] | None``.
```